compute_sdk/tests/unit/test_client.py

```diff
@@ -1102,3 +1102,16 @@ def test_ha_register_and_submit_warning_deduplication(gcc, mocker):
     assert len(records) == 2, "Only two distinct messages; warnings suppresses dups"
     assert arbitrary_hatext == str(records[0].message)
     assert "abcd" == str(records[1].message)
+
+
+def test_get_endpoints_with_role(gcc, randomstring):
+
+    gcc.get_endpoints(role="any")
+    gcc._compute_web_client.v2.get_endpoints.assert_called_with(role="any")
+
+    random_str = randomstring()
+    gcc.get_endpoints(role=random_str)
+    gcc._compute_web_client.v2.get_endpoints.assert_called_with(role=random_str)
+
+    gcc.get_endpoints()
+    gcc._compute_web_client.v2.get_endpoints.assert_called_with(role=None)
```

Comment:
Please consider parameterization.  For example:
```python
@pytest.mark.paramtrize("role", (None, randomstring_impl(), "any"))
def test_get_endpoints_with_role(gcc, role):
    gcc.get_endpoints()  # always check the "send nothing" case
    gcc._compute_web_client.v2.get_endpoints.assert_called_with(role=None)

    gcc.get_endpoints(role=role)
    gcc._compute_web_client.v2.get_endpoints.assert_called_with(role=role)
```

Reply:
Done.